lms/djangoapps/learner_home/serializers.py

```diff
@@ -460,7 +460,7 @@ def get_programs(self, instance):
 
     def get_credit(self, instance):
         """Pull credit statuses from context"""
-        credit_status = self.context["credit_statuses"].get(str(instance.course_id))
+        credit_status = self.context["credit_statuses"].get(instance.course_id)
```

Comment:
This is the root issue

Reply:
So, the dictionary is keyed off opaque-keys type? not strings huh.

Reply:
@schenedx , yes, I believe this is actually better practice since the opaque-key (or `CourseLocator`) covers several different keying methods for courses.

Historically, I think courses could follow either the format `/{org}/{course}/{run}` or our new `course-v1:{org}+{course}+{run}`. Using the string is fine but the opaque keys extract the info and convert all these to a standard data structure.